fanficfare/adapters/adapter_deviantartcom.py

```diff
@@ -0,0 +1,187 @@
+        logger.debug('Will now login to deviantARt as (%s)' % params['username'])
+
+        result = self.post_request(loginUrl, params, usecache=False)
+
+        if 'Log In | DeviantArt' in result:
+            logger.error('Failed to login to deviantArt as %s' % params['username'])
+            raise exceptions.FailedToLogin('https://www.deviantart.com', params['username'])
+        else:
+            return True
+
+    def requiresLogin(self, data):
+        return '</a> has limited the viewing of this artwork to members of the DeviantArt community only' in data
+
+    def isWatchersOnly(self, data):
+        return '<span>Watchers-Only Deviation</span>' in data
+
+    def requiresMatureContentEnabled(self, data):
+        return (
+            '>This content is intended for mature audiences<' in data
+            or '>This deviation is intended for mature audiences<' in data
+        )
+
+    def extractChapterUrlsAndMetadata(self):
+        isLoggedIn = False
+        logger.debug('URL: %s', self.url)
+
+        data = self.get_request(self.url)
+
+        soup = self.make_soup(data)
+
+        if self.requiresLogin(data):
+            if self.performLogin(self.url):
+                isLoggedIn = True
+                data = self.get_request(self.url, usecache=False)
+                soup = self.make_soup(data)
+
+        if self.isWatchersOnly(data):
+            raise exceptions.FailedToDownload(
+                'Deviation is only available for watchers.' +
+                'You must watch this author before you can download it.'
+            )
+
+        if self.requiresMatureContentEnabled(data):
+            # as far as I can tell deviantArt has no way to show mature
+            # content that doesn't involve logging in or using JavaScript
+            if not isLoggedIn:
+                self.performLogin(self.url)
+                isLoggedIn = True
+                data = self.get_request(self.url, usecache=False)
+                soup = self.make_soup(data)
+                if self.requiresMatureContentEnabled(data):
+                    raise exceptions.FailedToDownload(
+                        'Deviation is set as mature, you must go into your account ' +
+                        'and enable showing of mature content.'
+                    )
+
+        title = soup.select_one('h1').get_text()
+        self.story.setMetadata('title', stripHTML(title))
+
+        ## dA has no concept of status
+        # self.story.setMetadata('status', 'Completed')
+
+        pubdate = soup.select_one('time')['datetime']
+        self.story.setMetadata('datePublished', datetime.strptime(pubdate, '%Y-%m-%dT%H:%M:%S.%f%z'))
+
+        # do description here if appropriate
+
+        story_tags = soup.select('a[href^="https://www.deviantart.com/tag"] span')
+        if story_tags is not None:
+            for tag in story_tags:
+                self.story.addToList('genre', tag.get_text())
```

Comment:
I couldn't find a list more suitable to use for the tags than `genre`

Reply:
Depending on what the site generally uses, I believe adapters for some sites use `category`, some `genre` and some use a site specific list such as `freeformtags` or `sitetags` and let the user decide, possibly with a default setting to include it in `category` or `genre`.

Just depends on what makes the most sense for the common usage for that site.

Reply:
The tags on dA are entirely freeform, so I just picked genre as that seemed to be the most generic list I could use.